### PR TITLE
IOS integration

### PR DESCRIPTION
**Pull request by stelcheck**
Integration is not tested yet

### CODE REVIEW

**Comment by stelcheck** [comment]
Missing: Need a way to build it from CLI. Right now I do my builds through XCode, and I don't wanna have to do this.

File: platforms/ios/mage-sdk.xcodeproj/project.xcworkspace/xcshareddata/mage-sdk.xccheckout [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	<key>IDESourceControlProjectFavoriteDictionaryKey</key>
+	<false/>
+	<key>IDESourceControlProjectIdentifier</key>
+	<string>F72494E0-8686-48D7-AC5D-D59167A5AA34</string>
+	<key>IDESourceControlProjectName</key>
+	<string>mage-sdk</string>
+	<key>IDESourceControlProjectOriginsDictionary</key>
+	<dict>
+		<key>A208D586-A127-45FC-A6E4-0B5B18882F0C</key>
+		<string>ssh://github.com/stelcheck/mage-sdk-cpp.git</string>
```
Comment (ronkorving) [inline platforms/ios/mage-sdk.xcodeproj/project.xcworkspace/xcshareddata/mage-sdk.xccheckout]: careful (there's a few of these)

Comment (stelcheck) [inline platforms/ios/mage-sdk.xcodeproj/project.xcworkspace/xcshareddata/mage-sdk.xccheckout]: God damnit, I hate XCode.

Thanks for pointing it out, I will make sure to correct this.

**Comment by stelcheck** [comment]
Tested integration, seems to work properly so far.

**Comment by aogilvie** [comment]
You can build a Xcode project using (full customisation):
`xcodebuild ONLY_ACTIVE_ARCH=NO -workspace "${PROJECT_DIR}/${PROJECT_NAME}.xcworkspace" -scheme "${TARGET_NAME}" -configuration "${CONFIGURATION}" -sdk ${SF_OTHER_PLATFORM}${SF_SDK_VERSION} BUILD_DIR="${BUILD_DIR}" OBJROOT="${OBJROOT}" BUILD_ROOT="${BUILD_ROOT}" SYMROOT="${SYMROOT}" $ACTION`

Before executing `xcodebuild`, setup a target and scheme and all the relevant configuration in Xcode (iOS / OS X SDK etc.). I suggest a target for OS X and a target for iOS. It is also common to create a separate framework for simulator vs device (but we can do that later).

I don't mind doing this later today. It's pretty easy if one is used to playing with Xcode preferences.

**Comment by aogilvie** [comment]
I tried to run the Xcode project to build the framework and got the following error:

```
clang: error: no such file or directory: '/Users/Ally/Documents/Wizcorp/mage-sdk-cpp/vendor/libjson-rpc-cpp/src/jsonrpc/connectors/httpclient.cpp'
```

**Comment by stelcheck** [comment]
Did you git submodule update --init?

**Comment by stelcheck** [comment]
There is a lot of variables in the command you gave me, not sure how to actually use it

Also, I do not have a .xcworkspace file in my folder

**Comment by stelcheck** [comment]
The command given works for workspaces, this is an XCode project, however. This command won't work as is

**Pull request merged**